frontend/server/src/main/java/org/pytorch/serve/wlm/WorkLoadManager.java

```diff
@@ -150,6 +172,12 @@ public int getNumRunningWorkers(ModelVersionName modelVersionName) {
                 }
                 future.complete(HttpURLConnection.HTTP_OK);
             }
+
+            if (isRestartWorkers) {
+                logger.warn("restarting {} thread(s)", restartNumWorkers);
+                addThreads(threads, model, restartNumWorkers, future);
```

Comment:
Workers are restarted only when cpu_launcher is enabled (does not affect Torchserve when cpu_launcher_is not enabled). Restarting workers is useful/needed when user scales the number of workers up/down during execution. By restarting workers, launcher 1) re-allocates the cores that were pinned to killed workers in case of scale down; 2) avoids core overlap in case of scale up. It kills all existing workers and restarts scaled up/down number of workers. By doing so, launcher is configured properly when user scales the number of workers during execution.  I will add comments to the file to explain the logic.